### ISSUE TITLE
Improve navigation for Security section as a result of Security Checkup

### ISSUE THREAD

**Issue by daledupreez** [Automattic/wp-calypso#43144]
This issue stems from the discussion in #43101 around the behaviour of the nav bar due to the additional item added in that PR. The additional item makes the nav bar switch from a tab-like layout to a dropdown due to the default 720px width of the `Main` component in each of the security screens.

cc @fditrapani

#### Steps to reproduce

1. Starting at URL: `/me/security`
2. Enable the new Security Checkup feature via the `security/security-checkup` flag
3. See that the navigation shifts into a drop-down

#### What I expected
The items remain as a tab bar.

#### What happened instead
The navigation items are pushed into a dropdown component.

#dogfooding, #development

**Comment by daledupreez** [comment]
Sigh... I have _feelings_ about those "Back" buttons...

I also want to make sure that we don't hide key elements of this UI, but I don't actually know who all to pull in to comment on what is (or is not) important here. (There are also larger issues about the amount of love `/me/security` has seen in general.)

TLDR; I have thoughts about security, but I have no idea how valid they are!

**Comment by fditrapani** [comment]
> Sigh... I have feelings about those "Back" buttons...

Yes, there are some issues with it but this application is consistent with how it works everywhere else. 


> I also want to make sure that we don't hide key elements of this UI, but I don't actually know who all to pull in to comment on what is (or is not) important here.

The folks watching +a8csecops might have thoughts on these matter. I think in some ways this new UI brings more attention to the important aspects of account security which is why I was proposing leading the security section with it. What do you think about making the actions on the security scan more obvious:

![image](https://user-images.githubusercontent.com/6981253/84417899-4ebb1c80-abe4-11ea-88da-bd80580428a2.png)

**Comment by daledupreez** [comment]
I am a _big_ fan of making the actions much clearer!

I was actually also playing around with the idea of surfacing clear help for each item as much as possible. Here is a screen-cap of what I was playing with:
<img width="785" alt="Screenshot 2020-06-11 at 21 23 08" src="https://user-images.githubusercontent.com/3376401/84430489-0de5f680-ac2a-11ea-8bad-3241ccff3519.png">
This is still pretty rough, but was aimed at trying to make it easy for someone to find out more about all the rows in the table. As you can see, the layout doesn't look that good with the title-based alignment over multiple rows, and it also caused problems due to the current implementation using an `a` element around the entire row. (React doesn't like nested `a` elements, which I can get behind! :P)

As a side note, do you know how to pull in additional Material icons? I think that would help us here, especially if we want to distinguish between things that you can/should look into, versus things that you really should take care of. (As an example, showing that you last changed your password 5 years ago is _way_ more important than showing that you currently have 3 apps connected!)

**Comment by fditrapani** [comment]
Yeah, there's quite a lot going on there. I really like the idea of providing more clarity but perhaps, it might be better to offer that info on the subsequent screens. 

> especially if we want to distinguish between things that you can/should look into, versus things that you really should take care of. (As an example, showing that you last changed your password 5 years ago is way more important than showing that you currently have 3 apps connected!)

I noticed that too and tried using different icons for each section. I ultimately thought it was too busy, which made it hard to scan, and didn't like mixing things as you described. That's why I went down the route of moving the connected apps and social links to the tabbed menu. Moving those extra items out made the screen easier to read — the simpler check or warning icons also made it easier to see what you had to do. As for the password, based on what you said, I can see it living in the main content area.

**Comment by daledupreez** [comment]
Huh -- I was so focused on the Edit buttons I missed the shift of the informational items into separate tabs!

I did give this a bit more thought, and while using a button there would provide a much clearer action/CTA for each row, it would probably push many people to try and click the button instead of the row (and would require us fighting against the existing component to "make it so", unless we have the "button" there as a CTA, with no specific action on it).

I also think that pulling apart the most important items from the informational ones is _really_ helpful. I am just wondering whether it would still be useful to show them as a second, separate list on the main screen with clear wording to indicate that "this is stuff you might want to know". Going along with the icons you proposed above, these could use an `i` icon with a relatively neutral colour (likely a lighter grey than I have at present).

I'll work to split the lists as a first pass this morning, along with the copy changes, and then (hopefully) work on the nav/structure changes.

**Comment by fditrapani** [comment]
Splitting up the list sounds like a good idea. I can help find some appropriate icons for the second list and the copy too.

I wonder if we still need the tabbed navigation if we have all the screens represented in this list?

**Comment by daledupreez** [comment]
You may be right that we can leave off the tabs if the list _is_ the main navigation element. I'm exploring some of that right now (but don't have anything to show/share yet).
